File: LaboratoryWork/Laboratory_2/Kotlin/task3.kt
import java.util.Scanner
import kotlin.math.abs

fun is_number(a: Int): Boolean {
    /* Проверяет, что это число, а не цифра */
    return (a / 10 != 0)
}

fun cnt_odd_digits(a: Int): Int {
    /* Считает количество нечетных цифр в числе. */
    var res = 0
    var num = abs(a)  // Создаём новую переменную, так как параметр `a` неизменяемый

    while (num > 0) {
        res += (num % 2)
        num /= 10
    }

    return res
}

fun main() {
    val scanner = Scanner(System.`in`)  // Исправлено
    print("Введите количество чисел: ")
    val n = scanner.nextInt()

    for (i in 1..n) {
        print("Введите $i число: ")  // Используем строковый шаблон
        val num = scanner.nextInt()

        val cnt_odd = if (is_number(num)) {
            cnt_odd_digits(num)
        } else {
            println("$num - это цифра!")
            0
        }

        println("Количество нечетных цифр: $cnt_odd")  // Исправленный вывод
    }
}

File: LaboratoryWork/Laboratory_2/C++/task3.cpp
#include <iostream>
#include <cmath>

using namespace std;


bool is_number(int a) {
    /* Проверяет, что это число, а не цифра. */
    return (a / 10 != 0);
}


int cnt_odd_digits(int a) {
    /* Считает количество нечетных цифр в числе. */
    int res = 0;
    a = abs(a);

    while (a > 0) {
        res += (a % 2);
        a /= 10;
    }
    return res;
}


int main(void) {
    int n;
    cout << "Введите количество чисел: ";
    cin >> n;

    for (int i = 0; i < n; i++) {
        int num;
        cout << "Введите " << i + 1 << " число: ";
        cin >> num;

        int cnt_odd = 0;
        if (is_number(num)) {
            cnt_odd = cnt_odd_digits(num);
        } else {
            cout << num << " - это цифра!" << endl;
        }

        cout << "Количество нечетных цифр: " << cnt_odd << endl;
    }
}